### PR TITLE
TST: Re-enable test_moment_maps

### LINKED ISSUE #948: Moment map tests fail with spectrum1d cubes
**Reporter: Jesse Averbukh**

With the latest change to using Spectrum1D cubes in cubeviz, there is a bug that happens when running the moment map unit tests. They are currently inactive on main (after PR 547 is merged), so this work focuses on fixing those tests. The main culprit could be the spectrum1d_cube in conftest.py, but I am not positive about that. Traceback is posted in the following comment.

[🐱](https://jira.stsci.edu/browse/JDAT-1890)

 --- 

 DISCLAIMER: This issue was autocreated by the Jdaviz Issue Creation Bot on behalf of the reporter. If any information is incorrect, please contact [Duy Nguyen](https://github.com/duytnguyendtn)

### PR DESCRIPTION
**Pull request by pllim**
### Description

This pull request is to re-enable and fix a failing test.

Fixes #948 

Follow-up of #547

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [x] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [x] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [x] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Did the CI pass? If not, are the failures related?
- [x] Is a change log needed? If yes, is it added to `CHANGES.rst`?
- [x] Is a milestone set? Milestone is only currently required for PRs related to Imviz MVP.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

**Comment by pllim** [comment]
If I run the test code line by line in interactive session, I cannot reproduce this error! 🤯

**Comment by pllim** [comment]
I also noticed a lot of this:

`WARNING  specutils.spectra.spectrum1d:spectrum1d.py:205 Input WCS indicates that the spectral axis is not last. Reshaping arrays to put spectral axis last.`

I suspect there are some unnecessary data translation going on in the background but fixing that is out of scope here.

File: jdaviz/configs/cubeviz/plugins/moment_maps/tests/test_moment_maps.py [hunk 1]
```diff
@@ -24,4 +22,11 @@ def todo_fix_test_moment_calculation(spectrum1d_cube):
 
     assert mm.moment_available
     assert dc[1].label == 'Moment 0: test'
-    assert dc[1].get_object(cls=Spectrum1D, statistic=None).shape == (4, 2, 2)
+
+    # FIXME: Moment output has no WCS.
+    # Error from glue_astronomy/translators/spectrum1d.py to_object()
+    with pytest.raises(TypeError, match=r'data\.coords should be an instance of WCS'):
+        assert dc[1].get_object(cls=Spectrum1D, statistic=None).shape == (4, 2, 2)
```
Comment (pllim) [inline jdaviz/configs/cubeviz/plugins/moment_maps/tests/test_moment_maps.py:29]: I am not sure why the original test is expecting a cube back. Zeroth moment is the collapsed sum, so it should have been 2D.

Also, do you really expect the output of the moment map plugin to have WCS?

File: jdaviz/configs/cubeviz/plugins/moment_maps/tests/test_moment_maps.py [hunk 2]
```diff
@@ -1,27 +1,29 @@
...
-    mm.n_moment = 0
+    mm.n_moment = 0  # Collapsed sum, will get back 2D spatial image
     mm.vue_calculate_moment()
 
     assert mm.moment_available
     assert dc[1].label == 'Moment 0: test'
-    assert dc[1].get_object(cls=Spectrum1D, statistic=None).shape == (4, 2, 2)
+
+    result = dc[1].get_object(cls=CCDData)
+    assert_quantity_allclose(result, [[24, 30], [27, 33]] * u.Jy)
```
Comment (pllim) [inline jdaviz/configs/cubeviz/plugins/moment_maps/tests/test_moment_maps.py:29]: I have no idea if this result is actually correct. This is not the same as a simple sum along the relevant axis. Naively, I would have expected something like this:

```python
>>> a = np.arange(16).reshape((4, 2, 2))
>>> a.sum(axis=0)
array([[24, 28],
       [32, 36]])
```

**Comment by pllim** [comment]
Aww man, I didn't realize there is a remote data test... 😅

File: jdaviz/configs/cubeviz/plugins/moment_maps/tests/test_moment_maps.py (hunk 2 above)
Comment (pllim) [reply jdaviz/configs/cubeviz/plugins/moment_maps/tests/test_moment_maps.py:29]: Nevermind... trying to fix this broke Gaussian Smooth. I really don't grok why the shape keeps changing for cubes in this app.

File: jdaviz/configs/cubeviz/plugins/moment_maps/tests/test_moment_maps.py [hunk 3]
```diff
@@ -1,27 +1,30 @@
...
     mm.vue_calculate_moment()
 
     assert mm.moment_available
     assert dc[1].label == 'Moment 0: test'
-    assert dc[1].get_object(cls=Spectrum1D, statistic=None).shape == (4, 2, 2)
+
+    result = dc[1].get_object(cls=CCDData)
+    assert result.shape == (4, 2)
+
+    # FIXME: Need spatial WCS, see https://github.com/spacetelescope/jdaviz/issues/1025
```
Comment (pllim) [inline jdaviz/configs/cubeviz/plugins/moment_maps/tests/test_moment_maps.py:29]: xref #1025

**Review by kecnry — APPROVED** [review]

**Review by rosteen — APPROVED** [review]

**Pull request merged**